base_local_planner/src/speed_limiters/static_object_speed_limiter.cpp

```diff
@@ -0,0 +1,430 @@
+    max_allowed_linear_vel = max_linear_velocity_;
+    max_allowed_angular_vel = max_angular_velocity_;
+    if (params_.enabled == false || enabledFirmwareVersion_ == false)
+    {
+      if (test_data_ == nullptr || test_data_->enabledfirmwareVersion == false)
+      {
+        return true;
+      }
+    }
+
+    if (const char *env_var = std::getenv("ROSEVN_STATICOBJECTLIMITERENABLED"))
+    {
+      std::string val(env_var);
+
+      std::transform(val.begin(), val.end(), val.begin(), ::tolower);
+      if (val == "false")
+      {
+        return true;
+      }
+    }
+
+    if (params_.timeout > 0 && (ros::Time::now() - last_time_ < ros::Duration(params_.timeout)))
+    {
+      if (cachedMaxLinearVelocity_ > 0.0)
+      {
+        max_allowed_linear_vel = cachedMaxLinearVelocity_;
+      }
+      if (cachedMaxAngularVelocity_ > 0.0)
+      {
+        max_allowed_angular_vel = cachedMaxAngularVelocity_;
+      }
+      return true;
+    }
+    last_time_ = ros::Time::now();
+    cachedMaxLinearVelocity_ = max_linear_velocity_;
+    cachedMaxAngularVelocity_ = max_angular_velocity_;
+
+    // Cache off the current velocity so it can't be changed during processing
+    float currLinearVel = velocity_.linear;
+    float currAngularVel = fabs(velocity_.angular);
+    bool flipAngular = velocity_.angular < 0.0;
+
+    if (test_data_ != nullptr)
+    {
+      currLinearVel = test_data_->velocity.linear;
+      currAngularVel = test_data_->velocity.angular;
+      flipAngular = test_data_->velocity.angular < 0.0;
+    }
+
+    /*
+Get the CE stopping path edges to get the distance from the static object map. This would then test against the
+configured distances. Doing a non-linear mapping versus distance with more slowing the closer the static object is. This
+will be interesting as the slower the speed the tighter the region Smoothing speed will need to be done. Otherwise the
+Chuck will cycle between speeding up and slowing down.
+*/
+
+    tf::Stamped<tf::Pose> current_pose;
+
+    if (test_data_ == nullptr && !getCurrentPose(current_pose))
+    {
+      ROS_WARN_THROTTLE(1.0, "No pose in static object speed limiter");
+      return false;
+    }
+
+    int linearIndex, angularIndex = 0;
+    for (int i = 0; i < NUM_PLANE_LINEAR_SPEEDS; ++i)
+    {
+      if (LINEAR_SPEED_TO_INDEX_ARRAY[i] > currLinearVel)
+      {
+        break;
+      }
+      linearIndex = i;
+    }
+
+    for (int i = 0; i < NUM_PLANE_ANGULAR_SPEEDS; ++i)
+    {
+      if (ANGULAR_SPEED_TO_INDEX_ARRAY[i] > currAngularVel)
+      {
+        break;
+      }
+      angularIndex = i;
+    }
+
+    geometry_msgs::Point left, right;
+    srs::ChuckChassisGenerations::ChuckChassisType chassis = chassis_generation_;
+    if (test_data_ != nullptr)
+    {
+      chassis = test_data_->chassis_generation_;
+    }
+
+    // The plane tables indicate left right pairs of x,y coords
+    if (chassis == srs::ChuckChassisGenerations::ChuckChassisType::CHUCK5_PLUS)
```

Comment:
do chuck4's not get limited by this speed limiter ever?

Reply:
The intent of this speed limiter is to slow the Chucks down before the CE safety bubble detects the obstacle. Chuck 4s don't have CE safety boards to my knowledge